src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java

```diff
@@ -0,0 +1,164 @@
+package com.google.javascript.clutz;
+
+import com.google.javascript.jscomp.Compiler;
+import com.google.javascript.jscomp.SourceFile;
+import com.google.javascript.rhino.Node;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * If Clutz is running in incremental mode, closure can't resolve imported symbols, so it gives them
+ * names in the current namespace (ie `module$contents$current$file`), instead of names that
+ * reference the original namespace (ie `module$exports$imported$file`). ImportRenameMapBuilder
+ * builds a map from the local name to the exported name so the original names can be substituted.
+ */
+public class ImportRenameMapBuilder {
+
+  /**
+   * Build takes a collection of source files, parses them with the compiler, and walks the ast to
+   * find any `const variable = goog.require()` statements to build a map from the name closure
+   * generates in incremental mode to the exported name of the import.
+   */
+  public static Map<String, String> build(Compiler compiler, Collection<SourceFile> sourceFiles) {
+    Map<String, String> importRenameMap = new HashMap<>();
+    for (SourceFile sf : sourceFiles) {
+      Node ast = compiler.parse(sf);
```

Comment:
parsing is the most expensive operation in Closure Compiler. Can you double check that this doesn't actually parse the file again, and maybe find a way to make that more obvious in the code?

Reply:
We were reparsing, but I've fixed that.